### PR TITLE
Split changelog steps from code-freeze action

### LINKED ISSUE #33906: Code freeze automation failures on generation of changelog.
**Issue Description:**

The generation of the changelog automatically on release seems to have failed while attempting to remove the change files. See [this run](https://github.com/woocommerce/woocommerce/actions/runs/2671701840).

That issue needs to be fixed, and it'd be ideal if the release changelog generation could be run separate from the rest of the code freeze code, so that if it failed, it doesn't affect the notification.

### PR DESCRIPTION
**Pull request by jonathansadowski**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR splits the changelog generation steps from the code-freeze action into a new action and calls that action from the code-freeze action. This allows the code-freeze action to succeed if there's something preventing the generation of the changelog and allows us to re-run the changelog script stand-alone should the need arise.

Closes #33906 .

### How to test the changes in this Pull Request:

1. Fork this repository and merge this PR into trunk of your fork.
1. Create a release in your repository (any version x.y.z, doesn't really matter)
1. Make sure issues are enabled in your fork (to allow for milestones) from https://github.com/YOURFORK/woocommerce/settings
1. Make sure workflows are enabled in your fork from https://github.com/YOURFORK/woocommerce/actions
1. Make sure that workflows are allowed to read and write, as well as create PRs from https://github.com/YOURFORK/woocommerce/settings/actions
1. Add a GitHub token named `WC_BOT_TRIAGE_TOKEN` to your secrets (https://github.com/YOURFORK/woocommerce/settings/secrets/actions), containing a GitHub token that has `repo` scope. I wasn't able to get triggering one action from another to work with the `GITHUB_TOKEN` secret, it just silently failed.
1. Enable the code freeze action in your fork from https://github.com/YOURFORK/woocommerce/actions/workflows/release-code-freeze.yml
1. Run the code freeze action in your fork (from trunk) using the value 2022-08-18 for the time override. You probably want to skip the slack ping.
1. Ensure the action runs, creates milestones, and release branches as expected (based on your x.y release version)
1. Ensure that after the action runs, it runs the release-changelog.yml action, and creates two PRs to create the changelog and delete entries from trunk
1. Close those two PRs without merging and delete their branches
1. Run the release-changelog.yml action directly providing the previously created release branch and its version from https://github.com/YOURFORK/woocommerce/actions/workflows/release-changelog.yml
1. Ensure that again two PRs were created to create the changelog and delete entries from trunk

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by roykho — DISMISSED** [review]
Tested with my local fork and it works great! Good job!

File: .github/workflows/scripts/release-changelog.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 /**
- * Helper script for inserting NEXT_CHANGELOG contents into changelog and readme files.
+ * Helper script for inserting NEXT_CHANGELOG contents into changelog.
```
Comment (roykho) [inline .github/workflows/scripts/release-changelog.php]: I think we want to say `Helper script for inserting NEXT_CHANGELOG contents into readme.txt file.`

File: .github/workflows/scripts/release-changelog.php [hunk 2]
```diff
@@ -14,21 +14,17 @@
...
 $next_log  = preg_replace( "/= (\d+\.\d+\.\d+) YYYY-mm-dd =/", "= \\1 {$release_date} =", $next_log );
 
 // Convert PR number to markdown link.
 $next_log  = preg_replace( "/\[#(\d+)\]/", '[#$1](https://github.com/woocommerce/woocommerce/pull/$1)', $next_log );
-$changelog = preg_replace( "/^== Changelog ==\n\n/", "== Changelog ==\n\n{$next_log}\n", $changelog );
 
 $readme    = preg_replace( "/== Changelog ==\n(.*?)\[See changelog for all versions\]/s", "== Changelog ==\n\n{$next_log}\n\n[See changelog for all versions]", $readme );
 
 file_put_contents( $changelog_file, $changelog );
-file_put_contents( $readme_file, $readme );
```
Comment (roykho) [inline .github/workflows/scripts/release-changelog.php:34]: I think its the above line that should be removed?

Comment (jonathansadowski) [reply .github/workflows/scripts/release-changelog.php:34]: Good catch. It should've been the other one removed

File: .github/workflows/scripts/release-changelog.php (hunk 1 above)
Comment (jonathansadowski) [reply .github/workflows/scripts/release-changelog.php]: Yep, had some comments backward, fixed those, thanks.

**Review by roykho — APPROVED** [review]
Re-tested and LGTM! 🚢

**Pull request merged**